diff -up nut-2.8.0/clients/upsmon.c.rmpidf nut-2.8.0/clients/upsmon.c
--- nut-2.8.0/clients/upsmon.c.rmpidf	2022-04-23 13:56:06.000000000 +0200
+++ nut-2.8.0/clients/upsmon.c	2022-05-09 17:22:34.329547283 +0200
@@ -2314,6 +2314,7 @@ int main(int argc, char *argv[])
 
 	upslogx(LOG_INFO, "Signal %d: exiting", exit_flag);
 	upsmon_cleanup();
+	cleanpid(prog);
 
 	exit(EXIT_SUCCESS);
 }
diff -up nut-2.8.0/common/common.c.rmpidf nut-2.8.0/common/common.c
--- nut-2.8.0/common/common.c.rmpidf	2022-05-09 17:22:34.329547283 +0200
+++ nut-2.8.0/common/common.c	2022-05-09 17:26:48.671847245 +0200
@@ -358,6 +358,20 @@ pid_t parsepid(const char *buf)
 	return pid;
 }
 
+/* remove pid file */
+void cleanpid(const char *name)
+{
+       char    fn[SMALLBUF];
+
+       /* use full path if present, else build filename in PIDPATH */
+       if (*name == '/')
+               snprintf(fn, sizeof(fn), "%s", name);
+       else
+               snprintf(fn, sizeof(fn), "%s/%s.pid", PIDPATH, name);
+
+       unlink(fn);
+}
+
 /* open pidfn, get the pid, then send it sig
  * returns negative codes for errors, or
  * zero for a successfully sent signal
diff -up nut-2.8.0/include/common.h.rmpidf nut-2.8.0/include/common.h
--- nut-2.8.0/include/common.h.rmpidf	2022-05-09 17:22:34.329547283 +0200
+++ nut-2.8.0/include/common.h	2022-05-09 17:25:47.439740973 +0200
@@ -103,6 +103,9 @@ void chroot_start(const char *path);
 /* write a pid file - <name> is a full pathname *or* just the program name */
 void writepid(const char *name);
 
+/* remove pid file */
+void cleanpid(const char *name);
+
 /* parses string buffer into a pid_t if it passes
  * a few sanity checks; returns -1 on error */
 pid_t parsepid(const char *buf);
diff -up nut-2.8.0/server/upsd.c.rmpidf nut-2.8.0/server/upsd.c
--- nut-2.8.0/server/upsd.c.rmpidf	2022-04-23 13:56:07.000000000 +0200
+++ nut-2.8.0/server/upsd.c	2022-05-09 17:22:34.330547269 +0200
@@ -1515,6 +1515,7 @@ int main(int argc, char **argv)
 	ssl_cleanup();
 
 	upslogx(LOG_INFO, "Signal %d: exiting", exit_flag);
+	if (*pidfn) cleanpid(pidfn);
 	return EXIT_SUCCESS;
 }
 
